File: Examples/Node.js/managing-recognition/without-cloud-storage/read-barcode-from-localfile.js
var fs = require('fs');
var assert = require('assert');
var BarcodeApi = require('asposebarcodecloud');
var StorageApi = require('asposestoragecloud');

var configProps = require('../../data/config.json');
var data_path = '../../data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;

// ExStart:1
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

//Instantiate Aspose.BarCode API SDK
var barcodeApi = new BarcodeApi(config);

//Set the barcode file name 
var name = "sample-barcode.jpeg";

try {
		
	//invoke Aspose.BarCode Cloud SDK API to recognition of a barcode from file on server with parameters in body        
	barcodeApi.PostBarcodeRecognizeFromUrlorContent(null, null, null, null, null, data_path + name, function(responseMessage) {		
		assert.equal(responseMessage.status, 'OK');
		responseMessage.body.Barcodes.forEach(function(barcode) {
			console.log("Type :: " + barcode.BarcodeType);
			console.log("Codetext :: " + barcode.BarcodeValue);
		});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Ruby/managing-recognition/without-cloud-storage/read-barcode-from-localfile.rb
# Require asposecloudsdk

require 'aspose_barcode_cloud'

include AsposeStorageCloud
include AsposeBarcodeCloud
 
configFile = File.expand_path("../../../data/config.json", __FILE__);
configProps = JSON.parse( IO.read(configFile, encoding: 'utf-8') )
data_path = File.expand_path("../../../data/", __FILE__);
out_path = configProps['out_folder'];
  
#ExStart:1
AsposeApp.app_key_and_sid(configProps['api_key'], configProps['app_sid'])
AsposeBarcodeCloud::Configuration.debugging = false

#Instantiate Aspose.BarCode API SDK
@barcodeApi = BarcodeApi.new

#Set the image file url
name = 'sample-barcode.jpeg';

opts = { file: File.open(data_path + "/" << name,"rb"){ |io| io.read }}

#Invoke Aspose.BarCode Cloud SDK API to recognition of a barcode from file on server with parameters in body                                      
response = @barcodeApi.post_barcode_recognize_from_urlor_content(opts)

#print response
response.barcodes.each { |barcode| puts  barcode }

#ExEnd:1